backend/molgenis-emx2-tasks/src/main/java/org/molgenis/emx2/tasks/TaskServiceInDatabase.java

```diff
@@ -0,0 +1,332 @@
+package org.molgenis.emx2.tasks;
+
+import static org.molgenis.emx2.Column.column;
+import static org.molgenis.emx2.FilterBean.f;
+import static org.molgenis.emx2.FilterBean.or;
+import static org.molgenis.emx2.Operator.EQUALS;
+import static org.molgenis.emx2.Row.row;
+import static org.molgenis.emx2.TableMetadata.table;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.molgenis.emx2.*;
+import org.molgenis.emx2.sql.JWTgenerator;
+import org.molgenis.emx2.sql.SqlDatabase;
+
+public class TaskServiceInDatabase extends TaskServiceInMemory {
+  private Database database;
+  private String systemSchemaName = "ADMIN";
+
+  public TaskServiceInDatabase(String systemSchemaName) {
+    this.database = new SqlDatabase(false);
+    this.systemSchemaName = systemSchemaName;
+    this.init();
+  }
+
+  public TaskServiceInDatabase() {
+    this.database = new SqlDatabase(false);
+    this.init();
+  }
+
+  @Override
+  public String submit(Task task) {
+    // we insert task to database
+    save(task);
+
+    // and we add a change handler to write task status changes to database
+    task.setChangedHandler(
+        new TaskChangedHandler() {
+          @Override
+          public void handleChange(Task changedTask) {
+            save(changedTask);
+          }
+
+          @Override
+          public void handleOutputFile(Task changedTask, File outFile) {
+            saveWithOutput(changedTask, outFile);
+          }
+        });
+    return super.submit(task);
+  }
+
+  @Override
+  public Task getTask(String id) {
+    Task task = super.getTask(id);
+    if (task == null) {
+      return getTaskFromDatabase(id);
+    }
+    return task;
+  }
+
+  public Task getTaskFromDatabase(String id) {
+    final StringBuilder json = new StringBuilder();
+    database.tx(
+        db -> {
+          db.becomeAdmin();
```

Comment:
all callers get to be admin ?

Reply:
currently jobs can only be run by admins. However, I have some thoughts to make this more nuanced. In particular, we would like scripts to run in a particular (non admin) user when interacting with the database. Should double check if that is now not hidden (i.e., what token will be used?)